Added Angular SDK examples

engineering-this opened the pull request:
## CKEditor 4 Angular SDK examples
- Editor types, which presents `divarea` and `inline` editors.
- Component events and component attribute directives.
- Form example, with presentation of two-way binding.

## Todo
- Improve styling.
- Refactor styles to classes.
- Move styles to static file, so few deps could be dropped and webpack would run faster.

For now I didn't focus much on styling, as I'd rather hear out other people opinion.

## How to run
For some reason angular package doesn't work with linked package, so we need to create package locally and install it. Preserving symlinks in webpack doesn't solve it.

- You need to clone angular repo, build it and pack it:
```npm run build-package```
Inside `dist` run:
```npm pack```
- In `ckeditor-docs` install new deps from this branch.
- Install packed package:
```npm install location/to/built/tgz```

## Additional notes
Webpack doesn't work when two webpack config files are passed to grunt. After adding working webpack config to grunt both React and Angular build tasks doesn't work. As a workaround this I've added npm commands to build each integration, and forced grunt to use them.

This is second part to https://github.com/ckeditor/ckeditor-dev/issues/2920.

Closes ckeditor/ckeditor4-angular#14. Closes ckeditor/ckeditor-dev#2920.

f1ames commented on gruntfile.js:
```diff
@@ -66,9 +65,27 @@ module.exports = function( grunt ) {
-	grunt.registerTask( 'docs', [ 'api', 'fix-scayt-docs', 'prepare-examples', 'build-react', 'umberto' ] );
-	grunt.registerTask( 'docs-serve', [ 'api', 'fix-scayt-docs', 'prepare-examples', 'build-react', 'umberto', 'connect' ] );
+
+	function buildIntegrationTask( name ) {
```
Could you move this function to the end of the file?

engineering-this commented:
New changes:

- rebased,
- moved integrations into `scripts/integrations`,
- enabled IE polyfills,
- removed `dirty` and `touched` messages,
- moved styles to static html file and removed `.css` files from Angular source, co they aren't bundled by webpack, that allowed to remove some deps,
- updated styles (borrowed from React sample).

f1ames commented:
Rebased on latest `master`.

ma2ciek commented on scripts/integrations/angular/src/app/editor-types/editor-types.component.ts:
```diff
@@ -0,0 +1,24 @@
+	public editorData =
+		`<p>Hello world!</p>`;
+
+	inline = false;
```
```suggestion
	public inline = false;
```

ma2ciek commented on scripts/integrations/angular/src/app/events/events.component.ts:
```diff
@@ -0,0 +1,63 @@
+export class EventsComponent {
+	public isReadOnly = false;
+
+	isHidden = false;
```
Also, I'd mark these props as private/public or I'd not mark anything at all.

ma2ciek replied:
The rule can be enforced by the TSLint.

ma2ciek commented on scripts/integrations/angular/src/app/events/events.component.ts:
```diff
@@ -0,0 +1,63 @@
+
+	public componentEvents: ComponentEvent[] = [];
+
+	clearEventsLog() {
```
Also, this method is inconsistent with the next ones in the return type annotation.

ma2ciek commented on scripts/integrations/angular/src/app/events/events.component.ts:
```diff
@@ -0,0 +1,63 @@
+		this.componentEvents.length = 0;
+	}
+
+	saveEvent( eventName: string ): void {
```
This method can be marked as private and moved to the bottom of this file.

ma2ciek commented:
I added a few small comments to the TS files. We have not any style convention for them, but it would be good to have them styled the same way across the project.

ma2ciek commented:
And one thing which I discovered today is that in the CKEditor 5 integrations we have a few issues with multiple editors created with usage of the `*ngFor` directive - https://github.com/ckeditor/ckeditor5-angular/issues/86 - it'd be good to check it in the CKEditor 4 integration too (but for me it can be done in a follow-up).

f1ames commented:
@AnnaTomanek could you take a final look on this PR?

AnnaTomanek commented:
I added some corrections, I feel some stuff could still be improved.

The Events example contains no explanation text -- see the React sample for comparison/inspiration. Same for the Form example.

(minor) Some spacing could be added between the log and the button.

f1ames commented:
@engineering-this could you take a look at https://github.com/ckeditor/ckeditor-docs/pull/223#issuecomment-486622581 and try to add some descriptions for examples as suggested?

engineering-this commented:
Added short descriptions to samples.

f1ames commented:
Rebased on the latest `master`.

f1ames commented:
Some minor rewording.

@engineering-this The `@Input` is mentioned (as a link anchor) in the [description here](https://github.com/ckeditor/ckeditor-docs/pull/223/files#diff-3d09f6c821ff3ba42007d2eb4dfb5f04R4), however link points to `@Output` properties. Was it intended or just a typo?

engineering-this commented:
Just a silly mistake. I just fixed it.

f1ames commented:
Rebased on the latest `master`.
